ext/lib/mgmt/mcumgr/samples/smp_svr/README.md

```diff
@@ -0,0 +1,94 @@
+# smp_svr
+
+This sample application implements a simple SMP server with the following
+transports:
+    * Shell
+    * Bluetooth
+
+`smp_svr` enables support for the following command groups:
+    * fs_mgmt
+    * img_mgmt
+    * log_mgmt
+    * os_mgmt
+
+## Mynewt
+
+The Mynewt port of `smp_svr` is a regular Mynewt app.  The below example uses the nRF52dk as the BSP, so you may need to adjust accordingly if you are using different hardware.
+
+0. Create a Mynewt project and upload the Apache mynewt boot loader to your
+board as described here: http://mynewt.apache.org/latest/os/tutorials/nRF52/
+
+1. Add the mcumgr repo to your `project.yml` file:
+
+```
+repository.mynewt-mcumgr:
+    type: git
+    vers: 0-latest
+    url: 'git@github.com:apache/mynewt-mcumgr.git'
+```
+
+2. Create a target that ties the `smp_svr` app to your BSP of choice (nRF52dk in this example):
+
+```
+$ newt target create smp_svr-nrf52dk
+$ newt target set smp_svr-nrf52dk app=@mynewt-mcumgr/samples/smp_svr/apache \
+                                  bsp=@apache-mynewt-core/hw/bsp/nrf52dk    \
+                                  build_profile=debug
+```
+
+3. Build, upload, and run the application on your board:
+
+```
+$ newt run smp_svr-nrf52dk
+```
+
+## Zephyr
```

Comment:
Hi, I was not able to get this to work with nrf52_pca10040.

After flashing a bootloader and then smp_svr I get this in the logs:

```
[MCUBOOT] [INF] main: Starting bootloader
[MCUBOOT] [INF] boot_status_source: Image 0: magic=unset, copy_done=0xff, image_ok=0xff
[MCUBOOT] [INF] boot_status_source: Scratch: magic=unset, copy_done=0x0, image_ok=0xff
[MCUBOOT] [INF] boot_status_source: Boot source: slot 0
[MCUBOOT] [INF] boot_swap_type: Swap type: none
[MCUBOOT] [ERR] boot_go: bad image magic 0x0
[MCUBOOT] [ERR] main: Unable to find bootable image

```

Could you please post the steps to make this sample run along with the:

MCUBoot revision,
BOARD,
Zephyr revision (or just confirm that this PR branch was clean when tested)

that this was tested with?

NB: I built the bootloader with VALIDATE=NO, because I did not want to deal with signing at this point. When
building with VALIDATE=YES I got the message:

`[MCUBOOT] [ERR] boot_validate_slot: Image in slot 0 is not valid!`

Which I guess is expected when I have not signed the image.

Reply:
This was due to MCUBoot being broken. This is now fixed. https://github.com/runtimeco/mcuboot/issues/235